### ISSUE TITLE
fyne package uses 32bit msys2 compiler

### ISSUE THREAD

**Issue by 1Mochiyuki** [fyne-io/fyne#5070]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When using fyne package or fyne install, it will try to compile using the 32bit compiler

### How to reproduce

Install MSYS2
Open MSYS2 MINGW64 Shell
Run these 2 commands from [here](https://docs.fyne.io/started/)
```
pacman -Syu
pacman -S git mingw-w64-x86_64-toolchain
```

add msys264\mingw64\bin to Path System Environment Variable

run (if needed)
`go clean -cache`

run fyne package/fyne install

error:

```
C:\Program Files\Go\pkg\tool\windows_amd64\link.exe: running gcc failed: exit status 1
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/14.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: i386 architecture of input file `C:\Users\yuki\AppData\Local\Temp\go-link-1103428113\000000.o' is incompatible with i386:x86-64 output
collect2.exe: error: ld returned 1 exit status
```

### Screenshots

![image](https://github.com/user-attachments/assets/b71fe0db-01ec-4ca9-8a99-9465087d58d1) MSYS2 MING64 terminal used

![image](https://github.com/user-attachments/assets/e64138b2-5144-400a-9a0e-c8b0ddf85f46)
error received

![image](https://github.com/user-attachments/assets/8fcfd0a8-e0a1-4cc6-9100-bd83619503ef)
fyne and go compiler versions

![image](https://github.com/user-attachments/assets/fe5356a4-d399-4c2a-8612-8739e824622b) windows version

### Example code

`fyne package -icon "\path\to\icon\" --target windows `
`fyne install -icon "\path\to\icon\"`

both these commands dont work and will result in the error 

### Fyne version

2.5.0

### Go compiler version

1.22.5

### Operating system and version

Windows 11 Pro 23H2 Build 22631.3880

### Additional Information

I have uninstalled/reinstalled MSYS2 multiple times, cleaned go's cache multiple times

go build and run works without issues

**Comment by andydotxyz** [comment]
Attempted to correct the title, this does not seem to be about what compiler is used but somehow relates to resource bundling.

The question is why on this particular computer? Many, many, windows apps have been successfully packaged using the current tools.

**Comment by 1Mochiyuki** [comment]
im not sure, i dont think im doing anything wrong in particular/deviating from anything at all

**Comment by 1Mochiyuki** [comment]
not sure how this fixed it but apparently adding the --executable tag fixed it, no longer getting errors with it but i will get the error if i use it without --executable

**Comment by andydotxyz** [comment]
By specifying what executable to package you skip the build phase which would explain why it's not failing id guess.

Does your project have any .syso files which could be breaking the build?

**Comment by 1Mochiyuki** [comment]
it does not. it has a .bat file for running the package command though

**Comment by andydotxyz** [comment]
Can you please test with the hello world app (cmd/hello/ in the fyne repo)

**Comment by 1Mochiyuki** [comment]
yeah it seems to work with the hello world. and the icon stays and appears on the application and stays in task bar where on my other application it does not
![image](https://github.com/user-attachments/assets/b81e805b-1143-4c08-9e13-e8422f1691e0)

**Comment by andydotxyz** [comment]
Yes it seems like something in your project is bringing in a 32bit windows resource that is not triggered for the demo projects.
Can you please check your dependencies for such a file?

**Comment by 1Mochiyuki** [comment]
the only dependency that could affect this would be https://github.com/sqweek/dialog since i wanted to use a native file picker instead of using fyne's

**Comment by 1Mochiyuki** [comment]
just tested this using fyne-cross and it seems to not throw any errors. currently not on a windows machine so i cant test with --release at the moment

**Comment by andydotxyz** [comment]
That will be because fyne-cross is based in a Linux system that loads the windows toolchain. It does seem like something may be wrong with the C or Go compiler on your Windows setup, but I cannot see what that might be from the summary.
